src/main/sensors/gyroanalyse.c

```diff
@@ -12,24 +29,184 @@
+    fftBinStartCheck = fftBin(MIN_FILTER_FREQ); // first bin to check when looking for peak in frequency spectrum
+
+    arm_rfft_fast_init_f32(&fftInstance, FFT_WINDOW_SIZE);
 }
 
+/*
+ * Collect gyro data, to be analysed in gyroDataAnalyseUpdate function
+ */
 void gyroDataAnalyse(const gyroDev_t *gyroDev, const gyro_t *gyro)
 {
-    UNUSED(gyroDev);
     UNUSED(gyro);
+
+    static uint16_t index = 0;
+
+    if (gyroDataReady == false) {
+        gyroData[0][index] = gyroDev->gyroADC[0] * gyroDev->scale * fftWindow[index];
+        gyroData[1][index] = gyroDev->gyroADC[1] * gyroDev->scale * fftWindow[index];
+        gyroData[2][index] = gyroDev->gyroADC[2] * gyroDev->scale * fftWindow[index];
+        ++index;
+        if (index == FFT_WINDOW_SIZE) {
+            gyroDataReady = true;
+            index = 0;
+        }
+    }
 }
 
+void stage_rfft_f32(arm_rfft_fast_instance_f32 * S, float32_t * p, float32_t * pOut);
+typedef enum {
+    //STEP_ARM_RFFT_FAST_F32,
+    STEP_ARM_CFFT_F32,
+    STEP_STAGE_RFFT_F32,
+    STEP_ARM_CMPLX_MAG_F32,
+    STEP_ARM_MAX_F32,
+    STEP_COUNT
+} UpdateStep_e;
+
+/*
+ * Analyse gyro data using FFT, called periodically by the scheduler.
+ * This will execute once FFT_WINDOW_SIZE samples have been collected.
+ * For 8kHz gyro looptime and FFT_WINDOW_SIZE of 512, that is every 64ms, ie 15Hz
+ */
 void gyroDataAnalyseUpdate(timeUs_t currentTimeUs)
 {
+    static int axis = 0;
+    static int step = 0;
+    static float maxVal[3];
+    static uint32_t maxIdx[3];
     UNUSED(currentTimeUs);
+
+    if (gyroDataReady == false) {
+        // no new data, so return
+        return;
+    }
+    uint32_t startTime;
+    if (debugMode == DEBUG_FFT) {startTime = micros();}
+    arm_cfft_instance_f32 * Sint = &(fftInstance.Sint);
+    Sint->fftLen = fftInstance.fftLenRFFT / 2;
+
+    switch (step) {
+    /*case STEP_ARM_RFFT_FAST_F32:
+        // 217us
+        arm_rfft_fast_f32(&fftInstance, gyroData[axis], fftOut, 0);
+        // arm_rfft_fast_f32 is split into component parts see lib/main/DSP_lib/TransformFunctions/arm_rfft_fast_f32.c
+        break;*/
+    case STEP_ARM_CFFT_F32:
+        // 153us
+        arm_cfft_f32(Sint, gyroData[axis], 0, 1);
+        DEBUG_SET(DEBUG_FFT, 0, micros() - startTime);
+        break;
+    case STEP_STAGE_RFFT_F32:
+        // 58us
+        stage_rfft_f32(&fftInstance, gyroData[axis], fftOut);
+        DEBUG_SET(DEBUG_FFT, 1, micros() - startTime);
+        break;
+    case STEP_ARM_CMPLX_MAG_F32:
+        // 9us
+        arm_cmplx_mag_f32(fftOut, fftOut, fftBinCount);
+        DEBUG_SET(DEBUG_FFT, 2, micros() - startTime);
+        break;
+    case STEP_ARM_MAX_F32:
+        // 3us
+        arm_max_f32(&fftOut[fftBinStartCheck], fftBinCount - fftBinStartCheck - 1, &maxVal[axis], &maxIdx[axis]);
+        DEBUG_SET(DEBUG_FFT, 3, micros() - startTime);
+        DEBUG_SET(DEBUG_FFT_FREQ, axis, FFT_MAX_FREQ * (fftBinStartCheck + maxIdx[axis]) / fftBinCount);
+        break;
+    }
+
```

Comment:
Could keep a subset of the total fft in the frequency range of interest and do averaging over subsequent FFTs.  Might avoid changes caused a temporary event like prop wash.  The subset could be all 3 axis bins added together (or maybe max).